flaky tests/everest/test_everserver.py::test_status_contains_max_runtime_failure@starts_everest

https://github.com/equinor/ert/actions/runs/17210318777/job/48820212806?pr=11568

```
min_config = {'config_path': '.', 'controls': [{'max': 0.1, 'min': 0, 'name': 'my_control', 'type': 'well_control', ...}], 'forward_model': ['sleep 5'], 'install_jobs': [{'executable': '/usr/bin/sleep', 'name': 'sleep'}], ...}

    @pytest.mark.skip_mac_ci
    @pytest.mark.integration_test
    @pytest.mark.xdist_group(name="starts_everest")
    @pytest.mark.timeout(240)
    @patch("sys.argv", ["name", "--output-dir", "everest_output"])
    async def test_status_contains_max_runtime_failure(change_to_tmpdir, min_config):
        min_config["simulator"] = {
            "queue_system": {"name": "local", "max_running": 2},
            "max_runtime": 1,
        }
        min_config["forward_model"] = ["sleep 5"]
        min_config["install_jobs"] = [{"name": "sleep", "executable": which("sleep")}]
    
        config = EverestConfig(**min_config)
    
>       await wait_for_server_to_complete(config)

/home/runner/work/ert/ert/tests/everest/test_everserver.py:275: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/runner/work/ert/ert/tests/everest/test_everserver.py:68: in wait_for_server_to_complete
    wait_for_server(config.output_dir, 120)
/home/runner/work/ert/ert/src/everest/detached/client.py:144: in wait_for_server
    if server_is_running(*ServerConfig.get_server_context(output_dir)):
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/work/ert/ert/src/everest/config/server_config.py:64: in get_server_context
    server_info = ServerConfig.get_server_info(output_dir)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/work/ert/ert/src/everest/config/server_config.py:79: in get_server_info
    data = json.loads(json_string)
           ^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/json/__init__.py:346: in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/json/decoder.py:338: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.decoder.JSONDecoder object at 0x7f9e300804a0>, s = '', idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/json/decoder.py:356: JSONDecodeError
```

--- a/src/ert/services/_base_service.py
+++ b/src/ert/services/_base_service.py
@@ -12,6 +12,7 @@ from logging import Logger, getLogger
 from pathlib import Path
 from select import PIPE_BUF, select
 from subprocess import Popen, TimeoutExpired
+from tempfile import NamedTemporaryFile
 from time import sleep
 from types import FrameType
 from typing import TYPE_CHECKING, Any, Generic, Self, TypeVar
@@ -327,8 +328,10 @@ class BaseService:
             path = f"{self.service_name}_server.json"
 
         if isinstance(info, Mapping):
-            with open(path, "w", encoding="utf-8") as f:
-                json.dump(info, f, indent=4)
+            with NamedTemporaryFile(dir=f"{self._project}", delete=False) as f:
+                f.write(json.dumps(info, indent=4).encode("utf-8"))
+                f.flush()
+                os.rename(f.name, path)
 
         self._conn_info_event.set()
 

--- a/src/everest/config/server_config.py
+++ b/src/everest/config/server_config.py
@@ -1,5 +1,6 @@
 import json
 import os
+from pathlib import Path
 from typing import Any
 
 from pydantic import BaseModel, ConfigDict, Field, model_validator
@@ -71,19 +72,15 @@ class ServerConfig(BaseModel):
     @staticmethod
     def get_server_info(output_dir: str) -> dict[str, Any]:
         """Load server information from the hostfile"""
-        host_file_path = ServerConfig.get_hostfile_path(output_dir)
-        try:
-            with open(host_file_path, encoding="utf-8") as f:
-                json_string = f.read()
-
-            data = json.loads(json_string)
-            if not all(k in data for k in ("host", "port", "cert", "auth")):
-                raise RuntimeError("Malformed hostfile")
-        except FileNotFoundError:
-            # No host file
+        host_file_path = Path(ServerConfig.get_hostfile_path(output_dir))
+        if not host_file_path.exists():
             return {"host": None, "port": None, "cert": None, "auth": None}
-        else:
-            return data
+
+        data = json.loads(host_file_path.read_text(encoding="utf-8"))
+
+        if not all(k in data for k in ("host", "port", "cert", "auth")):
+            raise RuntimeError("Malformed hostfile")
+        return data
 
     @staticmethod
     def get_detached_node_dir(output_dir: str) -> str:

